### ISSUE TITLE
th:field outputs HTML entities in XML mode

### ISSUE THREAD

**Issue by EtienneMiret** [thymeleaf/thymeleaf#875]
Using

    <input th:field="*{foo}"/>

outputs

    <input id="foo" name="foo" value="&Eacute;tienne Miret"/>

when `foo` is bound to `Étienne Miret`.

This is of course not valid XML/XHTML.

Versions:
 * org.thymeleaf:thymeleaf-spring5 = 3.0.11.RELEASE
 * org.thymeleaf.extras:thymeleaf-extras-springsecurity5 = 3.0.4.RELEASE

Environment:
 * org.springframework:spring-framework-bom = 5.0.2.RELEASE
 * org.springframework.security:spring-security-bom = 5.0.0.RELEASE

Spring configuration:

    <bean id="templateResolver" class="org.thymeleaf.spring5.templateresolver.SpringResourceTemplateResolver">
      <property name="prefix" value="/WEB-INF/thymeleaf/"/>
      <property name="suffix" value=".xhtml"/>
      <property name="characterEncoding" value="UTF-8"/>
      <property name="templateMode" value="XML"/>
      <property name="cacheable" value="true"/>
    </bean>
  
    <bean id="templateEngine" class="org.thymeleaf.spring5.SpringTemplateEngine">
      <property name="templateResolver" ref="templateResolver"/>
      <property name="enableSpringELCompiler" value="true"/>
      <property name="additionalDialects">
        <set>
          <bean class="org.thymeleaf.extras.springsecurity5.dialect.SpringSecurityDialect"/>
        </set>
      </property>
    </bean>
  
    <bean id="viewResolver" class="org.thymeleaf.spring5.view.ThymeleafViewResolver">
      <property name="templateEngine" ref="templateEngine"/>
      <property name="contentType" value="application/xhtml+xml; charset=UTF-8"/>
    </bean>

Workaround: Don't use `th:field` and instead use `name` and `th:value`. The later outputs UTF-8 strings as expected.

I can provide a sample application on request.

**Comment by danielfernandez** [comment]
> `<input id="foo" name="foo" value="&Eacute;tienne Miret"/>`
>
> when foo is bound to Étienne Miret.
>
> This is of course not valid XML/XHTML.

This is not correct. The `&Eacute;` escape code is a perfectly valid XHTML entity, and could be valid XML if a DTD or Schema is specified that defines it. Given `th:field` is meant to be used in form tags like `<input>`, and these form tags are defined in XHTML's DTDs along with those HTML-equivalent escape codes, this cannot be considered a bug.
